contrib/objcwrapper/src/objc_generator.nit

```diff
@@ -65,169 +105,210 @@ class CodeGenerator
+		var fun_keyword = "fun"
+		if not opt_init_as_methods.value and method.is_init then
+			fun_keyword = "new"
 		end
-		file.write name
+
+		# Params
+		var params = new Array[String]
 		for param in method.params do
-			if param == method.params.first and not param.is_single then
-				file.write "(" + param.variable_name + ": " + type_convertor(param.return_type)
-			end
-			if param != method.params.first and not param.is_single then
-				file.write ", " + param.variable_name + ": " + type_convertor(param.return_type)
-			end
-			if param == method.params.last and not param.is_single then
-				file.write ")"
-			end
+			if param.is_single then break
+			params.add "{param.variable_name}: {param.return_type.to_nit_type}"
 		end
-		if method.return_type != "void" and not method.params.first.name.has("init") then
-			file.write ": " + type_convertor(method.return_type)
+
+		var params_with_par = ""
+		if params.not_empty then params_with_par = "({params.join(", ")})"
+
+		# Return
+		var ret = ""
+		if method.return_type != "void" and fun_keyword != "new" then
+			ret = ": {method.return_type.to_nit_type}"
 		end
+
+		file.write """
+{{{c}}}	{{{fun_keyword}}} {{{name}}}{{{params_with_par}}}{{{ret}}} in "ObjC" `{
+"""
 	end
 
-	private fun write_objc_init_call(classe_name: String, method: ObjcMethod, file: Writer)
+	# Write a combined call to alloc and to a constructor/method
+	private fun write_objc_init_call(class_name: String, method: ObjcMethod, file: Writer)
 	do
-		file.write """		return [[""" + classe_name + " alloc] "
+		# Method name and other params
+		var params = new Array[String]
 		for param in method.params do
 			if not param.is_single then
-				file.write param.name + ":" + param.variable_name
-				if not param == method.params.last then file.write " "
-			else
-				file.write param.name
-			end
+				params.add "{param.name}: {param.variable_name}"
+			else params.add param.name
 		end
-		file.write "];\n"
+
+		var c = method.comment_str
+
+		file.write """
+{{{c}}}		return [[{{{class_name}}} alloc] {{{params.join(" ")}}}];
+{{{c}}}	`}
+
+"""
 	end
 
 	private fun write_objc_method_call(method: ObjcMethod, file: Writer)
 	do
-		if method.return_type != "void" then file.write "return "
-		file.write "[self "
+		# Is there a value to return?
+		var ret = ""
+		if method.return_type != "void" then ret = "return "
+
+		# Method name and other params
+		var params = new Array[String]
 		for param in method.params do
 			if not param.is_single then
-				file.write param.name + ":" + param.variable_name
-				if not param == method.params.last then file.write " "
-			else
-				file.write param.name
-			end
+				params.add "{param.name}: {param.variable_name}"
+			else params.add param.name
 		end
-		file.write "];\n"
+
+		var c = method.comment_str
+
+		file.write """
+{{{c}}}		{{{ret}}}[self {{{params.join(" ")}}}];
+{{{c}}}	`}
+
+"""
 	end
 end
+
+redef class Text
+	# Nit equivalent to this type
+	private fun to_nit_type: String
```

Comment:
It made me notice that I slipped a `java` somewhere instead of `objc`... It is now fixed.